extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -127,6 +134,52 @@ BeanContainerListenerBuildItem build(
                 sslNativeConfig.isExplicitlyDisabled()));
     }
 
+    private static void validateBuildTimeConfig(final String datasourceName, final DataSourceBuildTimeConfig ds) {
+        //When the driver is not defined on the default datasource, we need to be more lenient as the datasource component might not be enabled at all:
+        if (!ds.driver.isPresent() && datasourceName == null) {
+            return;
+        }
+
+        String driverName = ds.driver.get();
```

Comment:
What if the driver for a **named** datasource is not set?

Reply:
if a named datasource is defined, it implies that there are some configuration lines for it.

Having a definition for a named datasource w/o the driver will throw an error; this is to be expected as it's an incomplete configuration.

We only allow for the default datasource to not have a driver, as there might be no intention to define a datasource (we decided early on that an extension being present but unconfigured is to be treated as "dormant").

Reply:
Well, an error right but not an `Optional` error.

If we think this case shouldn't happen, we should throw a proper error as the other you created.

Reply:
?
As I said, it's not an error to not have the driver and not have any other property: that means there is no unnamed datasource defined.

Reply:
Also remember this is _build time_ . Some additional checks need to be deferred to runtime.

Reply:
The issue is that your code would throw an exception when calling `String driverName = ds.driver.get();` in the case of a named datasource.

I don't think this is acceptable. I pushed an additional commit to fix it.

Reply:
ah, thanks!  I finally understand what you mean.